### ISSUE TITLE
GLError 1285 out of memory

### ISSUE THREAD

**Issue by Lurvelly** [Florian-Barthel/splatviz#44]
Hi, all

When using this command
```
python run_main.py
```
I get the GLError below:

Traceback (most recent call last):
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/OpenGL/latebind.py", line 43, in __call__
    return self._finalCall( *args, **named )
TypeError: 'NoneType' object is not callable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lurvelly/Workspace/splatviz/run_main.py", line 19, in <module>
    main()
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/lurvelly/Workspace/splatviz/run_main.py", line 14, in main
    splatviz.draw_frame()
  File "/home/lurvelly/Workspace/splatviz/splatviz.py", line 201, in draw_frame
    self.end_frame()
  File "/home/lurvelly/Workspace/splatviz/splatviz_utils/gui_utils/imgui_window.py", line 102, in end_frame
    self._imgui_renderer.render(imgui.get_draw_data())
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/imgui_bundle/python_backends/opengl_backend.py", line 237, in render
    gl.glBufferData(
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/OpenGL/latebind.py", line 63, in __call__
    return self.wrapperFunction( self.baseFunction, *args, **named )
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/OpenGL/GL/VERSION/GL_1_5.py", line 89, in glBufferData
    return baseOperation( target, size, data, usage )
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/OpenGL/latebind.py", line 47, in __call__
    return self._finalCall( *args, **named )
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/OpenGL/wrapper.py", line 700, in wrapperCall
    raise err
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/OpenGL/wrapper.py", line 693, in wrapperCall
    result = wrappedOperation( *cArguments )
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/OpenGL/platform/baseplatform.py", line 415, in __call__
    return self( *args, **named )
  File "/home/lurvelly/anaconda3/envs/gs-view/lib/python3.10/site-packages/OpenGL/error.py", line 230, in glCheckError
    raise self._errorClass(
OpenGL.error.GLError: GLError(
	err = 1285,
	description = b'out of memory',
	baseOperation = glBufferData,
	pyArgs = (
		GL_ARRAY_BUFFER,
		19440,
		c_void_p(95140738162736),
		GL_STREAM_DRAW,
	),
	cArgs = (
		GL_ARRAY_BUFFER,
		19440,
		c_void_p(95140738162736),
		GL_STREAM_DRAW,
	),
	cArguments = (
		GL_ARRAY_BUFFER,
		19440,
		c_void_p(95140738162736),
		GL_STREAM_DRAW,
	)
)

> I also tried this "export MESA_GL_VERSION_OVERRIDE=3.3", but doesn't work.

**Comment by Florian-Barthel** [comment]
Hi i have not seen this error before. On what kind of system are you running splatviz on? The error says "out of memory". Is it possible that the GPU or the CPU RAM is overloaded?

**Comment by Lurvelly** [comment]
> Hi i have not seen this error before. On what kind of system are you running splatviz on? The error says "out of memory". Is it possible that the GPU or the CPU RAM is overloaded?↳

Hi, Florian:

Thanks for your reply.

I attempted this on Ubuntu 24.04 running on a computer with an i7 8700K processor, 32 GB RAM, and a 1070 Ti GPU (8 GB) using NVIDIA-SMI 560.35.03, Driver Version 560.35.03, and CUDA Version 12.6. 

In my previous test, I tried rebooting my computer and didn't open any other programs—only ran splatviz—but it didn't work.↳

Today, I tried to reproduce the issue again, and it works without any errors. Recently, I upgraded some software on the system using sudo apt-get upgrade. I'm not sure if this upgrade fixed the issue.

![image](https://github.com/user-attachments/assets/b348f38e-b38e-4c11-b46f-828f1e42820a)

Best Regards
______________

Louis Li
